YuKitsune.Configuration.Env/YuKitsune.Configuration.Env/EnvConfigurationSource.cs

```diff
@@ -21,5 +24,30 @@ public override IConfigurationProvider Build(IConfigurationBuilder builder)
             EnsureDefaults(builder);
             return new EnvConfigurationProvider(this);
         }
+
+        /// <summary>
+        /// If no file provider has been set, for absolute Path, this will creates a physical file provider
+        /// for the nearest existing directory.
+        /// </summary>
+        public new void ResolveFileProvider()
+        {
+            if (FileProvider == null &&
+                !string.IsNullOrEmpty(Path) &&
+                System.IO.Path.IsPathRooted(Path))
+            {
+                string directory = System.IO.Path.GetDirectoryName(Path);
+                string pathToFile = System.IO.Path.GetFileName(Path);
+                while (!string.IsNullOrEmpty(directory) && !Directory.Exists(directory))
+                {
+                    pathToFile = System.IO.Path.Combine(System.IO.Path.GetFileName(directory), pathToFile);
+                    directory = System.IO.Path.GetDirectoryName(directory);
+                }
+                if (Directory.Exists(directory))
+                {
+                    FileProvider = new PhysicalFileProvider(directory, ExclusionFilters.System);
+                    Path = pathToFile;
+                }
+            }
+        }
```

Comment:
Not happy with this. The only _real_ difference from the base implementation is one line 47, but I'd prefer to have all of these preconditions here just in case.